packages/nimble-components/src/table/index.ts

```diff
@@ -461,44 +466,56 @@ export class Table<
         rowIndex: number,
         event: CustomEvent<TableActionMenuToggleEventDetail>
     ): Promise<void> {
-        let recordIds = event.detail.recordIds;
-        if (this.selectionMode !== TableRowSelectionMode.none) {
-            const row = this.table.getRowModel().rows[rowIndex];
-            if (row && !row.getIsSelected()) {
-                await this.selectSingleRow(rowIndex);
-            } else {
-                recordIds = await this.getSelectedRecordIds();
-            }
+        const selectionChanged = this.selectionManager.handleActionMenuOpening(
+            this.tableData[rowIndex]
+        );
+        if (selectionChanged) {
+            await this.emitSelectionChangeEvent();
         }
 
         this.openActionMenuRecordId = event.detail.recordIds[0];
-        const detail: TableActionMenuToggleEventDetail = {
-            ...event.detail,
-            recordIds
-        };
+        const detail = await this.getActionMenuToggleEventDetail(event);
         this.$emit('action-menu-beforetoggle', detail);
```

Comment:
It's not related to changes in this PR, but looking at this code again I think it's possible to get in a state where action-menu-toggle fires before action-menu-beforetoggle.

Or even that the action-menu is actually fully opened (fired before-toggle and toggle) before the table even fires action-menu-beforetoggle

It might not be an issue in practice unless action-menu-beforetoggle needs to be cancellable by clients. I think right now it's making the right trade-offs by having the table have the correct selection state before notifying the client that the menu opened, even if the menu is already rendered as open.

Not sure there is any action item here. Maybe it's just something else to consider in #1157. I can make a note there.